dds_web/scheduled_tasks.py

```diff
@@ -94,3 +94,50 @@ def set_available_to_expired():
                 )
                 for proj in errors[unit].keys():
                     scheduler.app.logger.error(f"Error for project '{proj}': {errors[unit][proj]} ")
+
+
+@scheduler.task("cron", id="expired_to_archived", hour=0, minute=1, misfire_grace_time=3600)
+def set_expired_to_archived():
+    """Search for expired projects whose deadlines are past and archive them"""
+    import sqlalchemy
+    from dds_web import db
+    from dds_web.database import models
+    from dds_web.errors import DatabaseError
+    from dds_web.utils import current_time
+    from dds_web.api.project import ProjectStatus
+
+    with scheduler.app.app_context():
+        archived_projs = db.session.query(models.ProjectStatuses.project_id).filter(
+            models.ProjectStatuses.status == "Archived"
+        )
+        expired_projs = models.ProjectStatuses.query.filter(
+            models.ProjectStatuses.project_id.not_in(archived_projs),
+            models.ProjectStatuses.status == "Expired",
+            models.ProjectStatuses.deadline <= current_time(),
+        ).all()
+
+        archive = ProjectStatus()
+        for proj_status in expired_projs:
+            new_status_row, delete_message = archive.archive_project(
+                project=proj_status.project,
+                current_time=current_time(),
+            )
+            proj_status.project.project_statuses.append(new_status_row)
+            scheduler.app.logger.debug(delete_message)
+
+            try:
+                db.session.commit()
+            except (sqlalchemy.exc.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
+                scheduler.app.logger.exception(err)
+                db.session.rollback()
+                raise DatabaseError(
+                    message=str(err),
+                    alt_message=(
+                        "Status was not updated"
+                        + (
+                            ": Database malfunction."
+                            if isinstance(err, sqlalchemy.exc.OperationalError)
+                            else ": Server Error."
+                        )
+                    ),
```

Comment:
Don't raise here, instead do what I suggested for the other cronjob - keep track of which fail with a dict and then print them out at the end.